File: frontend_reactnative_ts/services/auth.service.ts
import axios from './api';
import { User } from '../types';

// Login user
export const login = async (
  email: string,
  password: string
): Promise<{ token: string }> => {
  const res = await axios.post('/auth/login', { email, password });
  return res.data; // expected: { token }
};

// Register user
export const register = async (
  data: { profile?: string; username: string; email: string; password: string; role?: string }
): Promise<{ message: string; user: User }> => {
  const res = await axios.post('/auth/register', data);
  return res.data;
};

File: frontend_flutter/lib/services/auth_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:flutter_dotenv/flutter_dotenv.dart';

class AuthService {
  final String baseUrl = dotenv.env['BASE_URL'] ?? 'http://localhost:3000';

  Future<Map<String, dynamic>> login(String email, String password) async {
    final url = Uri.parse('$baseUrl/api/auth/login');

    final response = await http.post(
      url,
      body: {'email': email, 'password': password},
    );

    if (response.statusCode == 200) {
      final data = jsonDecode(response.body);
      return {'success': true, 'token': data['token'], 'user': data['user']};
    } else {
      final error = jsonDecode(response.body);
      return {'success': false, 'message': error['message'] ?? 'Login gagal'};
    }
  }

  Future<Map<String, dynamic>> register(
    String name,
    String email,
    String password,
  ) async {
    final url = Uri.parse('$baseUrl/api/auth/register');

    final response = await http.post(
      url,
      body: {'name': name, 'email': email, 'password': password},
    );

    if (response.statusCode == 201) {
      final data = jsonDecode(response.body);
      return {'success': true, 'token': data['token'], 'user': data['user']};
    } else {
      final error = jsonDecode(response.body);
      return {
        'success': false,
        'message': error['message'] ?? 'Registrasi gagal',
      };
    }
  }
}
